### PR TITLE
Versioning for Plugins

### LINKED ISSUE #190: `Feature`: Build/Version-Number in the GUI
**Describe the solution you'd like**
When the student-edition gets used we might get some (a lot of?) bug-reports or feature requests.
It would be quite convenient to display a build/version number somewhere so it'll be clear for everyone in which version the issue was encountered (hence there will always be someone who forgets to update their software).

Maybe we'll find a way to bundle the name of the tag of the release somewhere into the finished product.

### PR DESCRIPTION
**Pull request by dfuchss**
Show version in the Plugins. Bump Version to 2.0.0.

Fixes #190 

Add m2e to developer requirements

### CODE REVIEW

**Comment by dfuchss** [comment]
Products for Testing: https://github.com/kit-sdq/programming-lecture-eclipse-artemis/actions/runs/2287486356

**Comment by Shirkanesi** [comment]
Can the version-number be updated (somewhat) automatically from within eclipse? Or do we need to modify all those files on every update by hand?

**Comment by dfuchss** [comment]
It uses the Plugin Version. They have to be updated at any release (as in all Eclipse Products :D)

**Comment by dfuchss** [comment]
If you check for updates and install a new plugin version it will be shown within the Plugin :)

**Comment by Shirkanesi** [comment]
> It uses the Plugin Version. They have to be updated at any release (as in all Eclipse Products :D)

I ment for us as developers. Whether **we** can update this number in a convenient way :)

**Review by Shirkanesi — CHANGES_REQUESTED** [review]
lgtm (except the two annotations)

File: bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java [hunk 1]
```diff
@@ -347,6 +349,13 @@ private void createAssessmentTab(TabFolder tabFolder) {
 		this.addSelectionListenerForRefreshArtemisStateButton(btnRefreshArtemisState);
 		this.addControlToPossibleActions(btnRefreshArtemisState, Transition.ON_RESET);
 
+		var pluginVersion = Activator.getDefault().getBundle().getVersion();
+		Label version = new Label(assessmentComposite, SWT.NONE);
+		version.setLayoutData(new GridData(SWT.RIGHT, SWT.BOTTOM, true, false, 2, 1));
+		version.setAlignment(SWT.RIGHT);
+		version.setText("Artemis Grading " + //
```
Comment (Shirkanesi) [inline bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java:356]: String::format would be prettier :)

File: bundles/edu.kit.kastel.sdq.eclipse.student.view/src/edu/kit/kastel/eclipse/student/view/ui/ArtemisStudentView.java [hunk 2]
```diff
@@ -92,70 +93,77 @@ public void setFocus() {
...
+//        labelExercise.setText(EXERCISE);
+//        FontDescriptor boldDescriptor = FontDescriptor.createFrom(labelExercise.getFont()).setStyle(SWT.BOLD).setHeight(18);
+//        Font boldFont = boldDescriptor.createFont(labelExercise.getDisplay());
+//        labelExercise.setFont(boldFont);
+
+		var pluginVersion = Activator.getDefault().getBundle().getVersion();
+		Label version = new Label(exerciseGradingComposite, SWT.NONE);
+		version.setLayoutData(new GridData(SWT.LEFT, SWT.TOP, true, false, 2, 1));
+		version.setAlignment(SWT.RIGHT);
+		version.setText("Artemis Student Version: " + //
```
Comment (Shirkanesi) [inline bundles/edu.kit.kastel.sdq.eclipse.student.view/src/edu/kit/kastel/eclipse/student/view/ui/ArtemisStudentView.java:126]: see above

**Review by Shirkanesi — CHANGES_REQUESTED** [review]

File: bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java (hunk 1 above)
Comment (dfuchss) [reply bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java:356]: Yes , I'll fix that when I'm at the PC again.

**Pull request merged**